crates/metadata-server/src/local/server.rs

```diff
@@ -152,22 +152,88 @@ impl LocalMetadataServer {
 
         let mut nodes_configuration = StorageCodec::decode::<NodesConfiguration, _>(&mut bytes)?;
 
+        let mut modified = false;
+
         if let Some(node_config) =
             nodes_configuration.find_node_by_name(Configuration::pinned().common.node_name())
         {
-            if matches!(
+            // Only needed if we resume from a Restate version that has not properly set the
+            // MetadataServerState to Member in the NodesConfiguration.
+            if !matches!(
                 node_config.metadata_server_config.metadata_server_state,
                 MetadataServerState::Member
             ) {
-                // nothing to patch
-                return Ok(());
+                info!(
+                    "Setting MetadataServerState to Member in NodesConfiguration for node '{}'",
+                    node_config.name
+                );
+                let mut new_node_config = node_config.clone();
+                new_node_config.metadata_server_config.metadata_server_state =
+                    MetadataServerState::Member;
+
+                nodes_configuration.upsert_node(new_node_config);
+                modified = true;
             }
+        }
 
-            let mut new_node_config = node_config.clone();
-            new_node_config.metadata_server_config.metadata_server_state =
-                MetadataServerState::Member;
+        // If we have a node-id 0 in our NodesConfiguration, then we need to migrate it to another
+        // value since node-id 0 is a reserved value now. This can only be problematic if users are
+        // running multi node Restate cluster where there is a metadata-server node and others. In
+        // this situation if one of the other nodes was started with a forced node-id 0, then one
+        // needs to either unset this option or set it to the node-id that the migration step
+        // decided.
```

Comment:
Can we limit to a configuration that has only a single node? A configuration with multiple nodes + a node with id=0 is okay to just fail, or do you we expect that users might have configuration with more than one?

Reply:
I don't expect users to run multi-node setups with Restate < 1.2 even though it is technically possible (one metadata node and one worker node, for example). I am ok with failing if we detect that multiple nodes are contained in the `NodesConfiguration`.